pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTableBuilder.java

```diff
@@ -105,10 +105,16 @@ public void setColumn(int colId, @Nullable Object value)
       _currentRowDataByteBuffer.putInt(0);
       _variableSizeDataOutputStream.writeInt(CustomObject.NULL_TYPE_VALUE);
     } else {
-      int objectTypeValue = ObjectSerDeUtils.ObjectType.getObjectType(value).getValue();
+      ObjectSerDeUtils.ObjectType objectType = ObjectSerDeUtils.ObjectType.getObjectType(value);
+      int objectTypeValue = objectType.getValue();
       byte[] bytes = ObjectSerDeUtils.serialize(value, objectTypeValue);
+      //TODO: Remove this if clause we integrate vector type into the data table like BigDecimal
+      // Currently we are using int to store objectTypeValue but we don't add 4 to byte array length which messes up the vector deserialization
+      // Is this a bug? Should we fix it?
       _currentRowDataByteBuffer.putInt(bytes.length);
-      _variableSizeDataOutputStream.writeInt(objectTypeValue);
+      if (objectType != ObjectSerDeUtils.ObjectType.Vector) {
+        _variableSizeDataOutputStream.writeInt(objectTypeValue);
```

Comment:
@xiangfu0 @walterddr I found this weird case inside the codebase where we are putting `bytes.length` as the size but the actual value is `bytes.length + 4` cause we are also writing objectTypeValue to same buffer.

Is this a bug or am I missing something?